mtasa-blue/CODING_GUIDELINES.md

```diff
@@ -283,7 +283,18 @@ wherever possible.
 * Use the
 [noexcept](https://en.cppreference.com/w/cpp/language/noexcept_spec)
 specifier whenever possible, but **be cautious** as it can cause the
-program to terminate if an exception is thrown.
+program to terminate if certain conditions are met:
+- An exception is thrown in the noexcept function;
+- Noexcept function calls a **throwing** function;
+- Noexcept function calls a deallocation function that fails.
+
+<!--
+Throwing in noexcept function calls std::terminate.
+std::terminate calls std::terminate_handler which by default calls std::abort.
+std::abort raises SIGABRT signal.
+We don't handle that signal anywhere in the code, causing unavoidable termination of the process.
+Only the client code is somewhat protected from it by using the `SetUnhandledExceptionFilter` function.
+-->
```

Comment:
Why is this section hidden?

Reply:
Honestly I didn't think it would be needed much other than just a guidance.
If this information could be useful, I will unhide it

Reply:
It's up to you, I think by saying the program will terminate under these conditions it already implies that we don't handle the 'exiting' signal, but this adds some clarity as that may not be obvious to everyone.